src/engine/target.js

```diff
@@ -264,6 +332,103 @@ class Target extends EventEmitter {
         }
     }
 
+    /**
+     * Fixes up variable references in this target avoiding conflicts with
+     * pre-existing variables in the same scope.
+     * This is used when uploading this target as a new sprite into an existing
+     * project, where the new sprite may contain references
+     * to variable names that already exist as global variables in the project
+     * (and thus are in scope for variable references in the given sprite).
+     *
+     * If the given target has a block that references an existing global variable and that
+     * variable *does not* exist in the target itself (e.g. it was a global variable in the
+     * project the sprite was originally exported from), fix the variable references in this sprite
+     * to reference the id of the pre-existing global variable.
+     * If the given target has a block that references an existing global variable and that
+     * variable does exist in the target itself (e.g. it's a local variable in the sprite being uploaded),
+     * then the variable is renamed to distinguish itself from the pre-existing variable.
+     * All blocks that reference the local variable will be updated to use the new name.
+     // * @param {Target} target The new target being uploaded, with potential variable conflicts
```

Comment:
Confused about the commented out jsdoc here, it looks like it shouldn't need any params, but the unit tests pass in params?

Reply:
Oops, that was the remnant of a refactor I did. Fixed.